test/spec/modules/adlooxAnalyticsAdapter_spec.js

```diff
@@ -0,0 +1,230 @@
+
+        done();
+      });
+
+      it('should reject non-function options.toselector', function (done) {
+        const analyticsOptionsLocal = utils.deepClone(analyticsOptions);
+        analyticsOptionsLocal.toselector = esplode;
+
+        adapterManager.enableAnalytics({
+          provider: analyticsAdapterName,
+          options: analyticsOptionsLocal
+        });
+        expect(analyticsAdapter.context).is.null;
+
+        done();
+      });
+
+      [ 'client', 'clientid', 'platformid', 'tagid' ].forEach(function (o) {
+        it('should require options.' + o, function (done) {
+          const analyticsOptionsLocal = utils.deepClone(analyticsOptions);
+          delete analyticsOptionsLocal[o];
+
+          adapterManager.enableAnalytics({
+            provider: analyticsAdapterName,
+            options: analyticsOptionsLocal
+          });
+          expect(analyticsAdapter.context).is.null;
+
+          done();
+        });
+      });
+
+      it('should reject non-object options.params', function (done) {
+        const analyticsOptionsLocal = utils.deepClone(analyticsOptions);
+        analyticsOptionsLocal.params = esplode;
+
+        adapterManager.enableAnalytics({
+          provider: analyticsAdapterName,
+          options: analyticsOptionsLocal
+        });
+        expect(analyticsAdapter.context).is.null;
+
+        done();
+      });
+    });
+  });
+
+  describe('process', function () {
+    beforeEach(function() {
+      sandbox = sinon.sandbox.create();
+
+      sandbox.stub(events, 'getEvents').returns([]);
+
+      adapterManager.enableAnalytics({
+        provider: analyticsAdapterName,
+        options: utils.deepClone(analyticsOptions)
+      });
+
+      expect(analyticsAdapter.context).is.not.null;
+    });
+
+    afterEach(function () {
+      analyticsAdapter.disableAnalytics();
+      expect(analyticsAdapter.context).is.null;
+
+      sandbox.restore();
+      sandbox = undefined;
+    });
+
+    describe('event', function () {
+      it('should preload the script on AUCTION_END only once', function (done) {
+        const insertElementStub = sandbox.stub(utils, 'insertElement');
+
+        const uri = utils.parseUrl(analyticsAdapter.url(analyticsOptions.js));
+        const isLinkPreloadAsScript = function(arg) {
+          const href_uri = utils.parseUrl(arg.href);	// IE11 requires normalisation (hostname always includes port)
+          return arg.tagName === 'LINK' && arg.getAttribute('rel') === 'preload' && arg.getAttribute('as') === 'script' && href_uri.href === uri.href;
+        };
+
+        events.emit(EVENTS.AUCTION_END, auctionDetails);
+        expect(insertElementStub.calledWith(sinon.match(isLinkPreloadAsScript))).to.true;
+
+        events.emit(EVENTS.AUCTION_END, auctionDetails);
+        expect(insertElementStub.callCount).to.equal(1);
+
+        done();
+      });
+
+      it('should inject verification JS on BID_WON', function (done) {
```

Comment:
Should this test now be updated to just stub and validate the `loadExternalScript` function is called appropriately?

Now that the bid won handler is utilizing it?